src/SubscriptionContextFactory.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\SubscriptionContext;
+
+use Doctrine\Common\Annotations\AnnotationReader;
+use Doctrine\Common\Annotations\AnnotationRegistry;
+use Doctrine\Common\EventManager;
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\DriverManager;
+use Doctrine\ORM\Configuration;
+use Doctrine\ORM\EntityManager;
+use Doctrine\ORM\Mapping\Driver\XmlDriver;
+use Gedmo\Timestampable\TimestampableListener;
+
+class SubscriptionContextFactory {
+
+	private const DOCTRINE_CLASS_MAPPING_DIRECTORY = __DIR__ . '/../config/DoctrineClassMapping';
+
+	private array $config;
+	private Configuration $doctrineConfig;
+
+	private ?Connection $connection;
+	private ?EntityManager $entityManager;
+
+	public function __construct( array $config, Configuration $doctrineConfig ) {
+		$this->config = $config;
+		$this->doctrineConfig = $doctrineConfig;
+
+		$this->connection = null;
+		$this->entityManager = null;
+	}
+
+	public function getEntityManager(): EntityManager {
+		if ( $this->entityManager === null ) {
+			AnnotationRegistry::registerLoader( 'class_exists' );
+
+			$driver = new XmlDriver( self::DOCTRINE_CLASS_MAPPING_DIRECTORY );
```

Comment:
Please move the creation of the XML driver into a public `newMappingDriver` method and move the entity manager creation to test.